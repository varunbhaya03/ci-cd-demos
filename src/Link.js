import React from 'react'

const STATUS = {
  NORMAL: 'normal',
  HOVER: 'hover'    
}

class Link extends React.Component {
  state = {
    class: STATUS.NORMAL
  }

  render() {
    return (
      <a
        href={this.props.page || '#'}
        className={this.state.class}
        onMouseEnter={() => this.setState({class: STATUS.HOVER})}
        onMouseLeave={() => this.setState({class: STATUS.NORMAL})}
      >
        Learn React
      </a>
    )
  }
}
export default Link