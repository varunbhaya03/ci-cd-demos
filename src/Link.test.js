import React from 'react'
import Link from './Link'
import renderer from 'react-test-renderer'

test('Link color changes when hovered', () => {
  const component = renderer.create(
    <Link page="http://facebook.com">Facebook</Link>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  // manually trigger the callback
  tree.props.onMouseEnter()
  // re-rendering
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  // manually trigger the callback
  tree.props.onMouseLeave()
  // re-rendering
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

