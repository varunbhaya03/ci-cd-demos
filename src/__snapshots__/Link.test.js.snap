// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link color changes when hovered 1`] = `
<a
  className="normal"
  href="http://facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Learn React updates branch
</a>
`;

exports[`Link color changes when hovered 2`] = `
<a
  className="hover"
  href="http://facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Learn React updates branch
</a>
`;

exports[`Link color changes when hovered 3`] = `
<a
  className="normal"
  href="http://facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Learn React updates branch
</a>
`;
